jmix-flowui/flowui/src/main/java/io/jmix/flowui/sys/XmlInheritanceProcessor.java

```diff
@@ -0,0 +1,316 @@
+/*
+ * Copyright 2019 Haulmont.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.jmix.flowui.sys;
+
+import io.jmix.core.DevelopmentException;
+import io.jmix.core.Resources;
+import io.jmix.core.common.util.Dom4j;
+import io.jmix.core.common.util.ParamsMap;
+import org.apache.commons.lang3.StringUtils;
+import org.dom4j.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringWriter;
+import java.util.*;
+
+/**
+ * Provides inheritance of screen XML descriptors.
+ */
+@Component("flowui_XmlInheritanceProcessor")
+@Scope(BeanDefinition.SCOPE_PROTOTYPE)
+public class XmlInheritanceProcessor {
```

Comment:
`jmix-cuba` module contains test for `XmlInheritanceProcessor`: `com.haulmont.cuba.web.xmlinheritance.XmlInheritanceTest`. Please, provide the similar test for `jmix-flowui` module to make sure that everything works as expected.

Note that the above test is old fashioned, and currently we use Spock for tests. Please consider implementing tests using Stoke. Take a look as `view_registry.ViewRegistryTest` as an example. 

Also, consider adding test which checks that a view `title` is loaded correctly in case of extension.

Reply:
Found a spock test for the old `XmlInheritanceProcessor` here: `xml_inheritance.XmlInheritanceTest` -> maybe that's a better template for a new FlowUI test?

Reply:
added xml_inheritance.XmlInheritanceTest -> https://github.com/jmix-framework/jmix/pull/1172/commits/66525e56b6fcf7f440ba0b6df42ba4f40a980357

Reply:
also added test for correct view title, as you suggested.

Reply:
> Found a spock test for the old XmlInheritanceProcessor here: xml_inheritance.XmlInheritanceTest -> maybe that's a better template for a new FlowUI test?

this example definitely better